pkg/graph/resolvers/it_gov_task_statuses.go

```diff
@@ -1,12 +1,34 @@
 package resolvers
 
-import "github.com/cmsgov/easi-app/pkg/models"
+import (
+	"fmt"
+
+	"github.com/cmsgov/easi-app/pkg/apperrors"
+	"github.com/cmsgov/easi-app/pkg/models"
+)
 
 //NOTE these functions are deterministic. Ideally when implementing we should separate the logic which obtains any database information from the methods that calculate the status
 
 // IntakeFormStatus calculates the ITGovTaskListStatus of a system intake for the requester view
-func IntakeFormStatus(intake *models.SystemIntake) models.ITGovIntakeFormStatus {
-	return models.ITGISReady
+func IntakeFormStatus(intake *models.SystemIntake) (models.ITGovIntakeFormStatus, error) {
+	if intake.Step != models.SystemIntakeStepINITIALFORM {
+		return models.ITGISCompleted, nil
+	}
+	switch intake.RequestFormState {
+	case models.SIRFSNotStarted:
+		return models.ITGISReady, nil
+
+	case models.SIRFSInProgress:
+		return models.ITGISInProgress, nil
+
+	case models.SIRFSEditsRequested:
+		return models.ITGISEditsRequested, nil
+
+	case models.SIRFSSubmitted:
+		return models.ITGISCompleted, nil
+	default: //This is included to be explicit. This should not technically happen in normal use, but it is technically possible as the type is a type alias for string
+		return "", apperrors.NewInvalidEnumError(fmt.Errorf("invalid state provided for the Intake form state"), intake.RequestFormState, "SystemIntakeFormState")
```

Comment:
The error message here should probably be slightly different. This is a query, the user hasn't _provided_ any state, the problem is that invalid state has been stored. I think the error message should be something like "Intake <insert intake.ID> has an invalid value for its intake form state", or maybe "Intake <insert intake.ID> does not have a valid value for its intake form state".

Reply:
That's a good call, thanks for providing the phrasing, I will update it to use that!